### PR TITLE
key vault module in VNET

### PR DESCRIPTION
**Pull request by kristianschneider**
This locks the KV to a private endpoint and requires a VNET
https://app.zenhub.com/workspaces/mighty-ducks---the-outlaws-6193fe815d79fc0011e741b1/issues/energinet-datahub/the-outlaws/142

### CODE REVIEW

File: azure/key-vault/main.tf [hunk 1]
```diff
@@ -45,8 +45,35 @@ resource "azurerm_key_vault" "this" {
...
+   name                = "pe${lower(var.name)}${lower(var.project_name)}${lower(var.environment_short)}${lower(var.environment_instance)}"
+   location            = var.location
+   resource_group_name = var.resource_group_name
+   subnet_id           = var.private_endpoint_subnet_id
+   private_service_connection {
+     name                           = "psc${lower(var.name)}${lower(var.project_name)}${lower(var.environment_short)}${lower(var.environment_instance)}"
+     private_connection_resource_id = azurerm_key_vault.this.id
+     is_manual_connection           = false
+     subresource_names              = ["Vault"]
+  }
```
Comment (djorgensendk) [inline azure/key-vault/main.tf:64]: Same comments as the PR regarding service bus:
- introduction of module for azurerm_private_endpoint
- dash after pe and psc in the names
- where we can find documentation of the possible values for subresource_names

File: azure/key-vault/README.md [hunk 2]
```diff
@@ -29,6 +29,8 @@ This module creates the following resources.
 | `resource_group_name` | `string` | | **Required** | The name of the resource group in which to create the Key Vault. Changing this forces a new resource to be created. |
 | `location` | `string` | | **Required** | Specifies the supported Azure location where the resource exists. Changing this forces a new resource to be created. |
 | `sku_name` | `string` | | **Required** | The Name of the SKU used for this Key Vault. Possible values are `standard` and `premium`. |
+| `private_endpoint_subnet_id` | `string` | | **Required**  The ID of the private endpoint subnet
```
Comment (Renetnielsen) [inline azure/key-vault/README.md:32]: Add | after **Required** to push description to right box

File: azure/key-vault/README.md [hunk 3]
```diff
@@ -29,6 +29,8 @@ This module creates the following resources.
...
+| `private_dns_zone_name` | `string` | | **Required**   The name of the private dns zone
```
Comment (Renetnielsen) [inline azure/key-vault/README.md:33]: Add | after **Required** to push description to right box

File: azure/key-vault/main.tf [hunk 4]
```diff
@@ -14,14 +14,16 @@
 resource "null_resource" "dependency_setter" {
```
Comment (Renetnielsen) [inline azure/key-vault/main.tf:14]: Just remove this entire block, its no longer used. Dependencies does not seem to be a problem in TF 1+

**Review by Renetnielsen — CHANGES_REQUESTED** [review]
Great work, small comments nothing functional

**Pull request merged**